src/Kestrel.Core/Internal/Http2/Http2Stream.FeatureCollection.cs

```diff
@@ -246,13 +246,13 @@ MinDataRate IHttpMinResponseDataRateFeature.MinDataRate
 
         object IFeatureCollection.this[Type key]
         {
-            get => FastFeatureGet(key);
```

Comment:
Beware of exposing the underlying features directly. Any mutable fields should be reset per request.